src/client/csfle/state_machine.rs

```diff
@@ -181,6 +209,77 @@ impl CryptExecutor {
     }
 }
 
+#[derive(Debug)]
+struct Mongocryptd {
+    opts: MongocryptdOptions,
+    client: Client,
+    child: Mutex<Result<Process>>,
+}
+
+impl Mongocryptd {
+    async fn new(opts: MongocryptdOptions) -> Result<Self> {
+        let child = Mutex::new(Ok(Self::spawn(&opts)?));
+        let uri = opts.uri.as_deref().unwrap_or("mongodb://localhost:27020");
+        let mut options = crate::options::ClientOptions::parse_uri(uri, None).await?;
+        options.server_selection_timeout = Some(std::time::Duration::from_millis(10_000));
+        let client = Client::with_options(options)?;
+        Ok(Self {
+            opts,
+            client,
+            child,
+        })
+    }
+
+    async fn respawn(&self) -> Result<()> {
+        let mut child = match self.child.try_lock() {
```

Comment:
I'm probably missing something here, but what is the benefit of calling `try_lock` first and then `lock` if it fails? 
my assumption (which could be wrong) is that if the lock is free, `try_lock` and `lock` would both immediately return the lock. and that if it isn't free, we are going to end up calling `lock` regardless.

Reply:
If `try_lock` fails that tells us another task is attempting a respawn, so the failure branch doesn't need to call `Self::spawn`, it just needs to wait for the task holding the lock to finish and then return the result of that.  Just calling `lock` rather than `try_lock` would mean we can't distinguish between lock-immediately-acquired and lock-acquired-after-waiting.

Reply:
oh! I missed that we were just going ahead and returning here. that makes sense, thanks for the explanation